### ISSUE TITLE
Macro custom_graph! add generics support

### ISSUE THREAD

**Issue by Emoun** [Emoun/graphene#2]
Make the `custom_graph!` macro support generics:

Examples:
```
custom_graph!{
    struct SomeGraph
    where AdjListGraph<u32,()>
}
custom_graph!{
    struct NoEdgeWeightGraph<V>
    where AdjListGraph<V,()>
}
```

**Comment by Emoun** [comment]
Also need to handle lifetimes:

```
custom_graph!{
    pub struct G<'a>
    where AdjListGraph<'a,u32,()>
}
custom_graph!{
    pub struct G<'a,V,W>
    where AdjListGraph<'a,V,W>
}
```

**Comment by Emoun** [comment]
Landed complete generic/lifetime support in [v0.1.5](https://docs.rs/graphene/0.1.5/graphene/macro.custom_graph.html).
